tigerpath/majors_and_certificates/majors/ECO_2018.json

```diff
@@ -25,227 +25,261 @@
+          "course_list": [
+            "ECO 300",
+            "ECO 310"
+          ]
+        },
+        {
+          "name": "Macroeconomics",
+          "max_counted": 1,
+          "min_needed": 1,
+          "explanation": null,
+          "course_list": [
+            "ECO 301",
+            "ECO 311"
+          ]
+        },
         {
-          "name": "Statistics",
+          "name": "Economometrics",
           "max_counted": 1,
           "min_needed": 1,
-          "description": null,
-          "explanation": "",
+          "explanation": null,
           "course_list": [
-            "ECO 202",
-            "ORF 245"
+            "ECO 302",
+            "ECO 312"
           ]
         }
       ]
     },
     {
-      "name": "Track",
+      "name": "Economics",
+      "max_counted": 1,
+      "min_needed": 5,
+      "explanation": "In addition to the three core courses, concentrators must pass, on a graded basis, five other departmental courses. Departmentals can be any 300-, 400-, or 500-level Economics courses, or an approved cognate (see Cognates).\n\nStudents planning a senior thesis with empirical emphasis are strongly encouraged to take ECO 313; students planning a theoretical senior thesis are strongly encouraged to take ECO 317 and/or ECO 418.",
+      "req_list": [
+        {
+          "name": "Electives",
+          "max_counted": null,
+          "min_needed": 3,
+          "explanation": null,
+          "course_list": [
+            "ECO 3*",
+            "ECO 4*",
+            "ECO 5*"
+          ]
+        },
+        {
+          "name": "Cognates",
+          "max_counted": 2,
+          "min_needed": null,
+          "explanation": "Economics majors are allowed to count a maximum of two courses from other departments as cognates. These courses need to have substantial economics content. A course with an Economics cross-listing (indicated by an ECO 3XX, ECO 4XX, or ECO 5XX number in its first or second listing) counts as a regular departmental, not as a cognate. Courses that will be automatically recognized by SCORE as cognates are listed below. No application is necessary for these courses.",
+          "course_list": [
+            "COS 445",
+            "ELE 381",
+            "HIS 474",
+            "MAT 378",
+            "ORF 307",
+            "ORF 309",
+            "ORF 311",
+            "ORF 350",
+            "ORF 360",
+            "ORF 363",
+            "ORF 405",
+            "ORF 417",
+            "ORF 474",
+            "POL 335",
+            "POL 347",
+            "POL 349",
+            "POL 352",
+            "POL 385",
+            "WWS 309",
+            "WWS 340/PSY 321",
+            "WWS 373/CHV 373",
+            "WWS 408",
+            "WWS 466/HIS 467",
+            "WWS 524",
+            "WWS 538",
+            "WWS 582C",
+            "WWS 582F"
+          ]
+        }
+      ]
+    },
+    {
+      "name": "Tracks",
       "max_counted": 1,
       "min_needed": 1,

```

Comment:
This structure, with the redundant "Economics" track, technically works, though I agree that it's a bit odd. The structure you had before was nicer, but of course, it was causing issues.

The issue here is that the verifier currently doesn't bother checking requirements that don't contribute to anything (the thought when designing it was: well, why should it?), which is I guess why it gets left unchecked if you just do the simpler thing of promoting the PPE requirement and setting "max_counted" to 0.
I guess I didn't really anticipate having a completely useless, uproductive requirement (in the sense of not helping to satisfy the overall major) that should nevertheless still be checked, but thinking about it now, that doesn't seem as weird as I had thought, and so maybe it should be reevaluated.

One funny hack you could use is to set "max_counted" to some really small non-zero value like 0.000000000001, which is non-zero (forcing the verifier to check the requirement), but at the same time not large enough to make any difference whether it's satisfied or not.
It's definitely a weird hack, but it does the job of letting the verifier check the requirement but also not passing anything up.

So I would use that here and then drop the redundant Economics track. Perhaps also rename this to "Optional Tracks" to make it clear that it's optional.

It's important to properly comment the hack though, so that it's clear why it has such a weird value, and also so we know to remove it once we upgrade the verifier.

```suggestion
      "name": "Optional Tracks",
      # the following line is a hack to allow for an unproductive requirement that still gets checked. See discussion at PR #296 
      "max_counted": 0.000000000001,
      "min_needed": 0,
```